server.go

```diff
@@ -281,6 +287,9 @@ type serverPeer struct {
 	// The following chans are used to sync blockmanager and server.
 	txProcessed    chan struct{}
 	blockProcessed chan struct{}
+
+	reconnectMtx    sync.RWMutex
+	shouldReconnect bool
```

Comment:
Why do we need this _and_ the field we added in the prior commit? `ShouldDowngradeToV1`